### PR TITLE
Dialogue states should only be moveable if they have no connections attached

### PR DESCRIPTION
**Pull request by justinvdm**

### CODE REVIEW

**Comment by justinvdm** [comment]
Depends on #444

File: go/base/static/js/test/tests/campaign/dialogue/states/states.test.js [hunk 1]
```diff
@@ -332,36 +331,85 @@ describe("go.campaign.dialogue.states", function() {
...
+
+      it("should allow the state to be sorted if it isn't connected",
+      function() {
+        assert.deepEqual(
+          states.keys(),
+          ['state1','state2','state3','state4']);
+
+
+        $('[data-uuid="state3"] .titlebar')
+          .simulate('mousedown')
```
Comment (hodgestar) [inline go/base/static/js/test/tests/campaign/dialogue/states/states.test.js:369]: The other test (for connected states) doesn't have a mousedown. Is this one needed?

**Comment by hodgestar** [comment]
:+1: other than the small question about mousedown.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**